requirements.txt

```diff
@@ -14,3 +14,4 @@ filelock
 networkx
 jinja2
 fsspec
+z3-solver
```

Comment:
No, we shouldn't put this requirement here. If you need it in CI add to to our build.sh script.